ubuntu-advantage

```diff
@@ -313,10 +325,50 @@ is_livepatch_enabled() {
     canonical-livepatch status >/dev/null 2>&1
 }
 
-validate_user_pass_token(){
+validate_user_pass_token() {
     echo "$1" | grep -q '^[^:]\+:[^:]\+$'
 }
 
+check_url_with_credentials() {
+    local repo_url="$1"
+    local credentials="$2"
+    local file="$3"
+
+    local url
+    url=$(private_repo_url "$repo_url" "$credentials" "$file")
+
+    local tempfile log
+    tempfile="$(mktemp)"
+    log="$(mktemp)"
+    if "$APT_HELPER" download-file "$url" "$tempfile" >"$log" 2>&1; then
+        rm -f "$tempfile" "$log"
+        return 0
+    fi
+
+    local error_line
+    error_line=$(sed -n 's/^E: Failed to fetch [^ ]\+ \+//p' "$log")
+    if echo "$error_line" | grep -q '^\(HttpError\)\?401'; then
+        error_msg 'Invalid credentials'
+    else
+        error_msg "Failed checking credentials ($error_line)"
```

Comment:
We should use the term "token" instead of credentials, that's how we refer to the pair foo:bar in all other communications with the user. This could also be changed in the code (functions, variables) or else it might get confusing.

Reply:
done